### PR TITLE
`llama_supports_rpc()` function

### PR DESCRIPTION
**Pull request by martindevans**
Added `llama_supports_rpc` function to test for RPC support at runtime. This is useful for libraries such as LLamaSharp which need to check what the binaries were compiled with before trying to use certain features.

### CODE REVIEW

**Review by mofosyne — APPROVED** [review]

**Comment by slaren** [comment]
After #7640 we should consider including the RPC backend in all the llama.cpp builds by default. Then this function wouldn't be necessary.

File: llama.h [hunk 1]
```diff
@@ -413,6 +413,7 @@ extern "C" {
 
     LLAMA_API bool llama_supports_mmap       (void);
     LLAMA_API bool llama_supports_mlock      (void);
+    LLAMA_API bool llama_supports_rpc        (void);
```
Comment (ggerganov) [inline llama.h:416]:
```suggestion
    LLAMA_API bool llama_supports_rpc        (void); // TMP: https://github.com/ggerganov/llama.cpp/pull/7647#issuecomment-2140234367
```

**Comment by rgerganov** [comment]
We already have `ggml_cpu_has_rpc()` in `ggml.h`, does that work for you?

**Comment by martindevans** [comment]
Ah I hadn't noticed that GGML function. If that's equivalent to what I've added here that should be fine. LLamaSharp (and I would guess other wrappers) doesn't _usually_ expose GGML functions, but if this is only temporary anyway then it's fine.

**Comment by mofosyne** [comment]
@martindevans just following up on this PR as it was marked merge ready. Is the CI issue due to your code change, if not then resync against last known master with working ci commit.

**Comment by martindevans** [comment]
Well I managed to make a mess of that! 😨 

Is there still interest in merging this (i.e. a need for me to fix the mess)? The majority of the feedback seemed to be that this was going to be temporary, and wasn't even really needed since there's already an equivalent.

**Comment by rgerganov** [comment]
> The majority of the feedback seemed to be that this was going to be temporary, and wasn't even really needed since there's already an equivalent.

I don't think we need this function unless you have some good reasons for not using the existing `ggml_cpu_has_rpc()`.

**Comment by martindevans** [comment]
> unless you have some good reasons for not using the existing `ggml_cpu_has_rpc()`.

Just checking - it it exported in the C-API like all of the llama methods? if so, I think it'll work just fine for LLamaSharp to use that and I'll close this PR :)

**Comment by rgerganov** [comment]
yes, it is exported:
```bash
$ nm -D libllama.so | grep has_rpc
000000000018c130 T ggml_cpu_has_rpc

```

**Comment by martindevans** [comment]
Thanks for confirming, in that case I'll close this PR 👍

**Pull request closed without merging**